sql/hive/compatibility/src/test/scala/org/apache/spark/sql/hive/execution/HiveCompatibilitySuite.scala

```diff
@@ -1086,7 +1090,6 @@ class HiveCompatibilitySuite extends HiveQueryFileTest with BeforeAndAfter {
     "udf_sum",
     "udf_tan",
     "udf_tinyint",
-    "udf_to_byte",
```

Comment:
Can we change this behaviour? cc: @gatorsmile @dongjoon-hyun @wangyum 
If we do so, we need to update the guide, too:
https://github.com/apache/spark/blob/master/docs/sql-migration-guide-hive-compatibility.md
https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md

Reply:
This is within expectation. There are two cases in `udf_to_byte` that failed after this PR:
```
SELECT CAST(-129 AS TINYINT) FROM src tablesample (1 rows);
SELECT CAST(CAST(-1025 AS BIGINT) AS TINYINT) FROM src tablesample (1 rows);
```

The results will be null for Spark, while the result of HIve is not null.
I will change the migration guide if this looks ok to you.